src/Core/SpecificPrice/SpecificPriceFormatter.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\SpecificPrice;
+
+use Configuration;
+use Context;
```

Comment:
Core code SHOULD NOT depend on legacy classes. If you have legacy dependencies, then you MUST declare an interface and implement it as an Adapter. Also, try to keep legacy dependencies out of the interface.

Reply:
Thank you for your input @eternoendless I'll do that 👍

Reply:
So as we said, there is no adapter, a legacy class was created (it will never be used by the core, and no new code was added, it's just a legacy method being moved in a new legacy class due to unit testing issues).